### ISSUE TITLE
Sudden dips in val / inline inference metrics after resuming from a checkpoint

### ISSUE THREAD

**Issue by jpdunc23** [ai2cm/ace#483]
Recent ocean emulator runs show weird dips in validation / inference metrics after restarting from a checkpoint. In the screenshots below, the dips coincide with the epoch where the job resumed.

Possible explanations I can think of:
- There's a bug in the [branch](https://github.com/ai2cm/full-model/pull/1195/commits/62df788bcfefac839b56b96649f0c9bec8e76fe2) used for this experiment.
- Model buffers have different states before and after the restart.
- The optimizer state isn't saved / loaded correctly. 

![val_mean_loss_dip](https://github.com/user-attachments/assets/51dd77ff-d6df-4705-91fe-bbfac7217a98)

![inference_time_mean_rmse_dip](https://github.com/user-attachments/assets/5034c2d8-7544-4e16-81aa-292360f28cd5)

**Comment by mcgibbon** [comment]
Is this still an issue @jpdunc23 ?

**Comment by jpdunc23** [comment]
I think this is related to bug https://github.com/ai2cm/ace/issues/682 which will be fixed by https://github.com/ai2cm/ace/pull/819, so I set this issue to close automatically once that PR is merged.

This may be a case where the bug https://github.com/ai2cm/ace/issues/682 was actually helpful, though I haven't noticed this sort of behavior since. Typically I've seen this manifests as a spike in RMSE, not a drop.

**Comment by mcgibbon** [comment]
I have actually noticed this as a benefit in a recent run, leading me to suspect we would benefit from a lower EMA decay rate (which washes out less of the batch-to-batch variability). https://wandb.ai/ai2cm/ace/runs/0bb19gc0

The run restarted at batch 306780, the next epoch showing a large reduction in train/power_spectrum/mean_abs_norm_bias, and the epoch after that showing an even larger reduction (I think it also got pre-empted). This is variable specific, most get better but some get worse. FWIW `epoch_train_seconds` is a good proxy for pre-emption. There are three more pre-emptions in the run with similar behavior. However, while the first pre-emption doesn't have worse inference, the other three pre-emptions do. This kind of makes sense - the EMA is masking overfitting, and the overfitting will be worse with more training.

Also worth noting is, the pre-emption epochs where inference gets worse are the ones when the train power spectrum in the stratosphere gets worse. The one where T0 gets better has better inference.

This is getting me thinking that when we train a model with much more data and seem not to be overfitting, we should re-tune the decay rate @Arcomano1234 or even seeing what the model does without EMA.
